### PR TITLE
Fix a deprecation warning in map rotate

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -1730,7 +1708,8 @@ def plot(self, annotate=True, axes=None, title=True,
             tr = self._get_lon_lat(self.top_right_coord)
             x_range = list(u.Quantity([bl[0], tr[0]]).to(self.spatial_units[0]).value)
             y_range = list(u.Quantity([bl[1], tr[1]]).to(self.spatial_units[1]).value)
-            imshow_args.update({'extent': x_range + y_range})
+            imshow_args.update({'extent': x_range + y_range,
+                                'aspect': 1})
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: I didn't mean to commit this, it doesn't work.

Comment (nabobalis) [reply sunpy/map/mapbase.py]: Fix it?

**Comment by nabobalis** [comment]
Changelog too.

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -1030,35 +1031,12 @@ def rotate(self, angle=None, rmatrix=None, order=4, scale=1.0,
...
         """
         if angle is not None and rmatrix is not None:
-            raise ValueError("You cannot specify both an angle and a matrix")
+            raise ValueError("You cannot specify both an angle and a rotation matrix.")
         elif angle is None and rmatrix is None:
             rmatrix = self.rotation_matrix
 
-        # This is out of the quantity_input decorator. To allow the angle=None
-        # case. See https://github.com/astropy/astropy/issues/3734
-        if angle:
```
Comment (ayshih) [inline sunpy/map/mapbase.py]: For posterity, this is the line where the truth value of an array was being calculated

**Pull request merged**